### ISSUE TITLE
ID's of assettransformindex table are reset when you clear caches

### ISSUE THREAD

**Issue by metamachineco** [craftcms/cms#7267]
### Description
When I clear all caches using the `craft clear-caches/all` command, the rows in the `assettransformindex` table are deleted **and** the ID counter is reset.

But I'm using transforms in commerce email templates. As these emails are generated and sent to the client they only get the temporary url with the `transformId` and never the url to the transformed image.

So imagine this scenario: A commerce mail has been sent and the image path is 
`https://mysite.com/index.php?p=actions/assets/generate-transform&transformId=14
`
In the `assettransformindex` table, ID 14 refers to image1.jpg
If I now clear the caches, the `assettransformindex` table has been emptied and the ID's start counting from 1 again.

As the transforms are generated, the assettransformindex table is filling up again and it is now possible that ID 14 is referring to a completely other image.

**TL;DR**
I get ID 14 for a transform, then clear the table, and get 14 again, pointing to a new asset.

### Additional info

- Craft version: 3.5.15.1
- PHP version: 7.4.13
- Database driver & version: MySQL 5.7.31
- Plugins & versions: Commerce 3.2.11

**Comment by pdaleramirez** [comment]
@metamachineco The workaround for you issue is to set `generateTransformsBeforePageLoad` to true (https://craftcms.com/docs/3.x/config/config-settings.html#generatetransformsbeforepageload). This returns the asset URL location path instead of a transform id parameter.

**Comment by brandonkelly** [comment]
As of the next release, Craft will start setting that `generateTransformsBeforePageLoad` config setting to `true` before rendering emails, to ensure this sort of thing won’t happen going forward.

**Comment by brandonkelly** [comment]
Craft 3.5.17 is out now with that change.

**Comment by metamachineco** [comment]
Thanks Brandon!

**Comment by j-greig** [comment]
Help... I'm still having this problem in Commerce email templates and have a client chasing me very hard for a fix, is there anything else I can do? (see related issue above)

**Comment by brandonkelly** [comment]
@j-greig Will follow up over at craftcms/commerce#2034.
